odl/discr/lp_discr.py

```diff
@@ -279,15 +279,12 @@ def element(self, inp=None, **kwargs):
         elif inp in self.dspace:
             return self.element_type(self, inp)
 
-        # uspace element -> discretize
-        try:
+        if callable(inp):
+            # uspace element -> discretize
```

Comment:
This change should make many things much more easy to deal with, especially error messages.